src/FuncSharp.Examples/ITry-Parsing.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace FuncSharp.Examples
+{
+    public enum PersonParsingError
+    {
+        NameNotProvided,
+        AgeNotANumber,
+        AgeNegative,
+        AgeTooHigh,
+        HeightNotANumber,
+        HeightNegative,
+        HeightTooHigh
+    }
+
+    public class Person
+    {
+        private Person(string name, int age, decimal heightInCentimeters)
+        {
+            Name = name;
+            Age = age;
+            HeightInCentimeters = heightInCentimeters;
+        }
+
+        public string Name { get; }
+        public int Age { get; }
+        public decimal HeightInCentimeters { get; }
+
+        public static ITry<Person, IEnumerable<PersonParsingError>> Parse(string name, string age, string heightInCentimeters)
+        {
+            // You need to start with tries that have a collection as the type of error. So we MapError to a collection of strings here. But it can be a collection of any type.
+            // For validations, you would write your validation method, so it already returns IEnumerable as the error type and use MapError inside the method.
+            // Try.Aggregate method always executes all tries and then aggregates the success results or the errors. It doesn't stop on first error.
+            // You can transform the result into any type you want. This makes it nice and effective way of parsing values and combining the results.
+
+            ITry<string, IEnumerable<PersonParsingError>> parsedName = name.ToNonEmptyOption().ToTry(_ => new [] { PersonParsingError.NameNotProvided });
+            ITry<int, IEnumerable<PersonParsingError>> parsedAge = ParseAge(age);
+            ITry<decimal, IEnumerable<PersonParsingError>> parsedHeight =  ParseHeight(heightInCentimeters);
+
+            // You can aggregate more than 3 tries, just saving space here.
+            return Try.Aggregate(
+                parsedName,
+                parsedAge,
+                parsedHeight,
+                f: (n, a, h) => new Person(n, a, h)
+            );
+        }
+
+        private static ITry<int, IEnumerable<PersonParsingError>> ParseAge(string age)
+        {
+            var ageAsValidNumber = Try.Catch<int, Exception>(_ => Convert.ToInt32(age)).MapError(_ => new [] { PersonParsingError.AgeNotANumber });
+            var notNegativeAge = ageAsValidNumber.FlatMap(
+                a => a.ToOption().Where(aa => aa >= 0).ToTry(_ => new [] { PersonParsingError.AgeNegative })
```

Comment:
Looking at this I see boilerplate in the examples: `.ToOption().Where(aa => ...).ToTry(` and a missed use-case of FuncSharp. What about defining method `ToTry(predicate, error)` which would turn this example into much simpler:

```cs
var convertedAge = Try.Catch<int, Exception>(_ => Convert.ToInt32(age));
var numericAge = convertedAge.MapError(_ => PersonParsingError.AgeNotANumber);
var positiveAge = numericAge.FlatMap(age => age.ToTry(a => a >= 0, PersonParsingError.AgeNegative));
var reasonableAge = positiveAge.FlatMap(age => age.ToTry(a => a < 140, PersonParsingError.AgeTooHigh));
...
```

or actually, we could do even better and also implement `Where(predicate, otherwise)` with similar semantics to option.Where:

```cs
var convertedAge = Try.Catch<int, Exception>(_ => Convert.ToInt32(age));
var numericAge = convertedAge.MapError(_ => PersonParsingError.AgeNotANumber);
var positiveAge = numericAge.Where(a => a >= 0, PersonParsingError.AgeNegative);
var reasonableAge = positiveAge.Where(a => a < 140, PersonParsingError.AgeTooHigh);
...
```

@KaliCZ I'm happy to implement this, can you confirm that such methods would actually be usable in practice? At least for these examples, they'd really simplify it.

Reply:
I would definitely first document what we have and then go and start making changes to it. That way, we deliver some value now and if it shows that there are some issues, at least we already have some documentation done. Even if the code is not perfect.

Anyway, I was already thinking about this as well. 
- Where on ITry would definitely be useful. 
- ToTry with boolean condition would also definitely be useful. 
- Getting rid of the collection from the ITry is not so easy.

If you remember, we actually wanted to introduce a `Validation<T, E>` type that would be implement `ITry<T, IEnumerable<E>>`. This is needed for the aggregation as you still have only one type. 

Alternatively it would require us to make `Try.Aggregate` method that also requires a lambda to combine the errors into a collection of errors. I feel like having it automatically transform to a collection and then having a second overload where each try has a collection of errors and these are just concatenated might cause issues in C# compiler trying to figure out which one you want when you actually already have a collection..